app/controllers/posts_controller.rb

```diff
@@ -5,6 +5,36 @@ def index
   end
 
   def show
-    @post = Post.find(params[:id])
+    @post = User.find(params[:user_id]).posts.find(params[:id])
+  end
+
+  def new
+    @user = User.find(params[:user_id])
+    @post = @user.posts.new
+    render :new, locals: { post: @post }
+  end
+
+  def create
+    @user = User.find(params[:user_id])
+    add_post = @user.posts.new(post_params)
```

Comment:
- [ ] As pointed out by the previous reviewer, kindly make sure that a post is created on behalf of the currently logged-in user(current_user) of the app, that's `current_user.posts.new(post_params)`.  You really need to call the `current_user `method previously created in the ApplicationController when creating a post.

Reply:
Hello thank you for the suggestions. I have implemented

Reply:
This change has been done well